### PR TITLE
Various changes and additions requested by Polina

### PR DESCRIPTION
**Pull request by WhatisRT**
I hope I didn't miss anything. I'll copy the relevant changes to the multicurrency spec tomorrow, so don't merge it yet.

### CODE REVIEW

File: goguen/formal-spec/transactions.tex [hunk 1]
```diff
@@ -90,6 +90,8 @@ \section{Transactions}
   option will additionally allow users to write their own programmatic solutions
   to choosing for-fees inputs.
 
+  \item $\TxId$ is always computed from values of type $\Tx$, never from $\ShelleyTx$ or $\GoguenTx$.
```
Comment (polinavino) [inline goguen/formal-spec/transactions.tex]: I don't think it's the best place for this, since the explanation is about `txid` the function (which always hashes the original serialization).

File: goguen/formal-spec/transactions.tex [hunk 2]
```diff
@@ -269,18 +267,7 @@ \section{Transactions}
...
-
-\begin{figure*}[htb]
-  \begin{align*}
-    \Nothing_d ~\in~& \Data \\
-    & \text{the Data representation of empty type} \\
-  \end{align*}
-  \caption{Empty Data Representation}
-  \label{fig:defs:todata}
-\end{figure*}
+is structural equality between them.
```
Comment (polinavino) [inline goguen/formal-spec/transactions.tex:268]: I posted in the Plutus channel asking about whether it should be the same type or structural equality - I am not really sure tbh.

File: shelley-mc/formal-spec/utxo.tex [hunk 3]
```diff
@@ -233,18 +233,18 @@ \subsection{UTxO Transitions}
       ~
       \\
       \mathsf{adaID}~\notin \dom~{\fun{forge}~tx} \\
-      \forall txout \in \txouts{txb}, ~ \fun{getValue}~txout  ~\geq ~ \epsilon \\~
+      \forall txout \in \txouts{txb}, ~ \fun{getValue}~txout  ~\geq ~ 0 \\~
+      \forall txout \in \txouts{txb}, ~ \fun{getCoin}~txout ~\geq \\
+      \fun{valueSize}~(\fun{getValue}~txout) * \fun{minUTxOValue} pp \\~
```
Comment (polinavino) [inline shelley-mc/formal-spec/utxo.tex]: is the `valueSize` calculation defined anywhere? I am not sure if it needs to be in the formal spec or not. Maybe ask in the rule-based-spec channel? 
It should really be `scaledValueSize`, or equivalently
`(\fun{valueSize}~(\fun{getValue}~txout) * \fun{minUTxOValue} pp ) / (size (the Coin type)`

The function `valueSize` should be something like
```(the number of PolicyIDs) * (size PolicyID) + (the total number of AssetIDs) * (size (Quantity) + maxSize AssetID) ...```

File: goguen/formal-spec/utxo.tex [hunk 4]
```diff
@@ -951,6 +950,8 @@ \subsection{Updating the UTxO State}
       \\
       \mathsf{adaID}~\notin \dom~{\fun{forge}~tx} \\
       \forall txout \in \txouts{txb}, ~ \fun{getValue}~txout  ~\geq ~ 0 \\~
+      \forall txout \in \txouts{txb}, ~ \fun{getCoin}~txout ~\geq \\
+      \fun{valueSize}~(\fun{getValue}~txout) * \fun{minUTxOValue} pp \\~
       \\
       \fun{txsize}~{tx}\leq\fun{maxTxSize}~\var{pp} \\
       \fun{txexunits}~{txb} \leq \fun{maxTxExUnits}~{pp}
```
Comment (polinavino) [inline goguen/formal-spec/utxo.tex:965]: We should have a call about what to do about ExUnits, I think tomorrow. I am going to schedule it. I think (based on discussions in the formal and rule-based spec meetings), we are going back to per-script budgets, which requires quite a bit of rule changes. Eg. , the `SVAL` transition type is no longer needed. But we still need a max memory threshold in addition to the `Steps` and `Memory-Steps` budget.

File: goguen/formal-spec/properties.tex [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  when processing the transaction is that the inputs marked for paying
+  fees are moved to the fee pot.
+\item If a Shelley transaction is accepted, it is fully processed.
+\item If a transaction extends the UTxO, all its scripts validate, and
+  if it has a script that does not validate, it cannot extend the
+  UTxO.
+\item A valid transaction that does not forge tokens satisfies the
+  accounting property of the Shelley ledger where the type $\Coin$ is
+  replaced by $\Value$.
+\end{itemize}
```
Comment (polinavino) [inline goguen/formal-spec/properties.tex:24]: (i) is true for ALL tokens, both custom and Ada. The way that the accounting property is formulated takes into consideration the `forge` field, which balances out any tokens that the transaction creates or destroys 
(v) I dont think we should refer to properties of other ledgers

(vi) all ada should be fungible with other ada
- this property is an emergent property of the ledger because it is a result of Ada not being forged, the accounting property holding for each token type, and the ledger translation of Shelley -> Goguen making only the right kind of Ada tokens, something like `(AdaPolicy, AdaTokenName, q)`

File: shelley-mc/formal-spec/utxo.tex (hunk 3 above)
Comment (WhatisRT) [reply shelley-mc/formal-spec/utxo.tex]: I named `valueSize` in analogy to `txSize`, and I didn't see an implementation for that at that time, but now I've seen that it's in an Appendix of the Shelley spec. Reading that, I've realized that there is also an issue with `txSize`, because we actually need a function like `valueSize` to compute `txSize`. I guess we should make an Appendix for that.

Scaling the size makes sense, otherwise the semantics of `minUTxOValue` changes after Shelley, so I'll do that.

File: goguen/formal-spec/properties.tex (hunk 5 above)
Comment (WhatisRT) [reply goguen/formal-spec/properties.tex:24]: (i) was an attempt to combine property (i) and (iii) from https://github.com/input-output-hk/cardano-ledger-specs/issues/1284
(v) I don't know about that. It would of course be a difficult property to prove (especially formally), but it provides a strong guarantee that Goguen doesn't break anything.
(vi) The issue I have here is that it isn't a property of the STS, it's a property of the STS and its initial conditions. So we could say something like: Assuming the initial state of the ledger was the result of a translation from Shelley, all tokens with `AdaPolicy` have `AdaTokenName`.

Comment (polinavino) [reply goguen/formal-spec/properties.tex:24]: these are all meant to be properties of a valid ledger state, derived with relevant rules from a valid initial state, so (vi) is relevant. 

(v) yea - leave it unless someone else complains

(i) i would rephrase a bit, 
"whether it is fully processed or just paying fees. In particular, this implies that the total
  amount of Ada in the system is constant."

Comment (WhatisRT) [reply goguen/formal-spec/properties.tex:24]: I've made some changes in the most recent commit to address this, so I'll mark this as resolved

File: shelley-mc/formal-spec/utxo.tex (hunk 3 above)
Comment (WhatisRT) [reply shelley-mc/formal-spec/utxo.tex]: After noticing the issues with `txSize` in Shelley vs Goguen, I don't think we can scale it with the size of `Coin`, instead we have to use the size of a `Value` that only contains Ada.

File: goguen/formal-spec/transactions.tex (hunk 1 above)
Comment (WhatisRT) [reply goguen/formal-spec/transactions.tex]: I'm not sure where else to put it though. In the vicinity of `txid`, it's obvious from the type of `txid`.

File: shelley-mc/formal-spec/utxo.tex (hunk 3 above)
Comment (polinavino) [reply shelley-mc/formal-spec/utxo.tex]: We should use the size of Coin, but scale the ada-only case of CompactValue by the scaling factor (size of Coin)

File: goguen/formal-spec/transactions.tex (hunk 1 above)
Comment (polinavino) [reply goguen/formal-spec/transactions.tex]: Move it to 4.1, before the note after the bullet points (before "Note that when submitting a transaction, ..")

**Review by polinavino — APPROVED** [review]
